cc/crypto/server_encryptor.cc

```diff
@@ -91,7 +91,7 @@ absl::Status ServerEncryptor::InitializeRecipientContexts(const EncryptedRequest
 
   // Create recipient contexts.
   absl::StatusOr<RecipientContext> recipient_context =
-      SetupBaseRecipient(serialized_encapsulated_public_key, server_key_pair_, kOakHPKEInfo);
+      recipient_context_generator_.GenerateRecipientContext(serialized_encapsulated_public_key);
```

Comment:
While this goes slightly beyond the scope of this PR, I have questions about the contexts.

a) Why are they lazily generated in the Decrypt call? (And what about Encrypt? Why not there?)
b) Are you expecting this to be called in a multithreaded context? Right now this is thread-unsafe.
c) How expensive are the contexts? Could they be just created in Decrypt for the duration of the call, meaning you wouldn't need to keep this state around at all and the whole class could just be devolved into a few free functions?

Reply:
It can only be initialized in the decrypt method, because the client sends a key needed for initialization in the first method.

Also, this initialization is encapsulated in the decrypt and not the create method, because otherwise the app developers will need to extract the key themselves and put it in the create method. So this implementation the implementation details from them.